### PR TITLE
Fix http message

### PR DESCRIPTION
**Pull request by lihuiba**
fix operator priority bug in http/status.cpp;
enhance estring;

### CODE REVIEW

File: common/estring.cpp [hunk 1]
```diff
@@ -55,27 +55,34 @@ size_t estring_view::find_last_not_of(const charset& set) const
...
 {
     uint64_t ret = 0;
     for (unsigned char c : *this) {
-        if (c >= '0' && c <= '9') {
-            ret = ret * 16 + (c - '0');
-        } else if (c >= 'A' && c <= 'F') {
-            ret = ret * 16 + (c - 'A' + 10);
-        } else if (c >= 'a' && c <= 'f') {
-            ret = ret * 16 + (c - 'a' + 10);
+        unsigned char cc = c - '0';
```
Comment (lihuiba) [inline common/estring.cpp:73]: reducing branches in machine code

File: common/estring.cpp [hunk 2]
```diff
@@ -55,27 +55,34 @@ size_t estring_view::find_last_not_of(const charset& set) const
 
 bool estring_view::to_uint64_check(uint64_t* v) const
 {
-    v ? (*v = 0) : 0;
+    uint64_t val = 0;
```
Comment (lihuiba) [inline common/estring.cpp:58]: simplify the syntax

File: net/http/message.cpp [hunk 3]
```diff
@@ -331,7 +331,7 @@ int Request::redirect(Verb v, estring_view location, bool enable_proxy) {
     }
     StoredURL u(location);
     auto new_request_line_size = verbstr[v].size() + sizeof(" HTTP/1.1\r\n") +
-        enable_proxy ? full_url_size(u) : u.target().size();
+        (enable_proxy ? full_url_size(u) : u.target().size());
```
Comment (lihuiba) [inline net/http/message.cpp:334]: oops...

File: net/utils.cpp [hunk 4]
```diff
@@ -101,22 +101,25 @@ int _gethostbyname(const char* name, Delegate<int, IPAddr> append_op) {
...
+
+inline __attribute__((always_inline))
+void base64_translate_3to4(const char *in, char *out)  {
     static const unsigned char tbl[] =
         "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
     auto v = htonl(*(uint32_t *)in);
-    auto x = *(xlator *)(&v);
-    *(uint32_t *)out = ((tbl[x.a] << 24) + (tbl[x.b] << 16) +
-                        (tbl[x.c] << 8) + (tbl[x.d] << 0));
+    auto x = (xlator*) &v;
```
Comment (lihuiba) [inline net/utils.cpp:118]: to eliminate a warning

**Review by beef9999 — APPROVED** [review]

**Pull request merged**